compiler/src/iree/compiler/API2/Embed.h

```diff
@@ -128,9 +128,34 @@ IREE_EMBED_EXPORTED void ireeCompilerSessionGetFlags(
 // compiler.
 //===----------------------------------------------------------------------===//
 
+enum iree_compiler_diagnostic_severity_t {
+  IREE_COMPILER_DIAGNOSTIC_SEVERITY_NOTE = 0,
+  IREE_COMPILER_DIAGNOSTIC_SEVERITY_WARNING = 1,
+  IREE_COMPILER_DIAGNOSTIC_SEVERITY_ERROR = 2,
+  IREE_COMPILER_DIAGNOSTIC_SEVERITY_REMARK = 3,
+};
+
 IREE_EMBED_EXPORTED iree_compiler_invocation_t *ireeCompilerInvocationCreate(
     iree_compiler_session_t *session);
 
+// Enables a callback to receive diagnostics. This is targeted at API use of
+// the compiler, allowing fine grained collection of formatted diagnostic
+// records. It is not completely identical to
+// |ireeCompilerInvocationEnableConsoleDiagnostics| which produces output
+// suitable for an interactive stream (including color detection, etc) and has
+// additional features for reading source files, etc. This implementation makes
+// no demands on the system state.
```

Comment:
OOC this doesn't place any restrictions on `callback` just saying that this function itself doesn't make any demands?

Reply:
Reworded to say what I meant.